lib/jsCompiler.js

```diff
@@ -18,6 +19,7 @@ module.exports = function (app, callback) {
   const logger = require('./tools/logger')(app.get('params').logging)
   const fsr = require('./tools/fsr')(app)
   let preprocessor
+  let gitignoreFiles = gitignoreScanner(path.join(app.get('appDir'), '.gitignore'))
```

Comment:
Don't run the gitignore scanner until just before it's needed, that way it only gets run when the compiler is enabled and ready to go.